### ISSUE TITLE
Any Tool over 5s fails in official hosted version 

### ISSUE THREAD

**Issue by MartianInGreen** [mckaywrigley/chatbot-ui#1450]
The maximum length of API calls when using tools seems to be only 5s in the hosted version, I believe it's related to this issue: https://stackoverflow.com/questions/68771480/nextjs-vercel-504-error-function-invocation-timeout

This is the exact error:
```
An error occurred with your deployment

FUNCTION_INVOCATION_TIMEOUT
```

**Comment by mckaywrigley** [comment]
Will investigate!

**Comment by mckaywrigley** [comment]
@MartianInGreen Yeah so looks like you'd just need to upgrade plans with Vercel. The code is running on their servers so they have to charge for compute time.

**Comment by MartianInGreen** [comment]
I'm not hosting it myself, it's the version hosted under https://www.chatbotui.com/

**Comment by mckaywrigley** [comment]
> I'm not hosting it myself, it's the version hosted under https://www.chatbotui.com/

Oh yikes. For some reason I'm not running into the issue. Let me test some things...

**Comment by MartianInGreen** [comment]
Mh okay, if I can do anything to help let me know. I've tested both with and without VPN and with multiple location, two devices and two different API and consistently get the error.

**Comment by jkmerg** [comment]
![image](https://github.com/mckaywrigley/chatbot-ui/assets/136703631/3e97d8c8-d87d-4518-88a2-8859ca8e3cd3)

Me too! This is my schema tool:

{
  "openapi": "3.0.0",
  "info": {
    "title": "Web Content Scraper",
    "description": "A tool to scrape the text content from a web page",
    "version": "1.0.0"
  },
  "servers": [
    {
      "url": "https://api.openai.com/v1"
    }
  ],
  "paths": {
    "/scrape": {
      "post": {
        "summary": "Scrape web page content",
        "operationId": "scrapeWebContent",
        "requestBody": {
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "type": "object",
                "properties": {
                  "url": {
                    "type": "string",
                    "description": "The URL of the web page to scrape"
                  }
                },
                "required": ["url"]
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "Successful response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/WebContent"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "WebContent": {
        "type": "object",
        "properties": {
          "title": {
            "type": "string",
            "description": "The title of the web page"
          },
          "content": {
            "type": "string",
            "description": "The text content of the web page"
          }
        }
      }
    }
  }
}
